### ISSUE TITLE
Prompt curators to add AE 'infective ability' to metagenotype annotations

### ISSUE THREAD

**Issue by CuzickA** [pombase/canto#2121]
https://github.com/PHI-base/phipo/issues/167

Currently we are capturing the 'pathogenicity/virulence' PHI-base high level phenotypes using the Annotation Extension 'infective ability'.

Kim H-K and I thought that it would be a good idea to prompt the curator to enter this field for all metagenotype annotations. We may need a new term for 'not tested' or 'unknown' in some cases. We don't want curators to overlook this field as we may miss capturing key information enabling users to perform wide data searches for our high level phenotype terms.

**Comment by jseager7** [comment]
@kimrutherford Is it actually possible to make an annotation extension mandatory? Currently, if an annotation extension is 'required' (meaning the cardinality of the extension doesn't include zero), Canto still lets you continue an annotation without adding an extension.

I've checked this with one of our _assayed_using_ annotation extensions that has a cardinality of 2 (since it's supposed to be used to record the genes / proteins involved in a protein-protein interaction), but Canto just lets you continue without adding any extensions. This doesn't seem valid, since it means the extension behaves like the cardinality range is {0, 2}.

**Comment by jseager7** [comment]
@CuzickA I'm not sure it makes sense to add a phenotype term for an 'unknown' or 'untested' change in infective ability: assuming you're regarding a phenotype as some observable quality that is expressed, surely an unknown or untested quality isn't observable?

Also, if you regard 'an unknown amount of change in some quality' as a phenotype, then where do you draw the line? It would logically follow that you would need an option for 'unknown change' in every single quantifiable phenotype, and that implies needing to curate everything that _isn't_ known, which isn't feasible.

I think it would be better to (somehow) always prompt the user to add an 'infective ability' term, but only _where applicable_, meaning the user still has the ability to bypass the extension if they need to. We should then regard any lack of 'infective ability' extension as implying that the 'infective ability' is unknown or untested.

**Comment by ValWood** [comment]
Yes we should do this if the cardinality of a term is 1 or more, definitely.

This would be really helpful; for when users use the copy/edit facility and change the term, they are not currently prompted for an extension and this is often omitted.
However, in this situation, we would ideally have a way to prompt for non-mandatory extensions too.

**Comment by kimrutherford** [comment]
> Is it actually possible to make an annotation extension mandatory?

There's no way at the moment.  We'd need to change the long annotation work-flow to force adding the mandatory extensions.  And we'd need to change the annotation edit dialog to disable the "OK" button if a mantatory extension is missing.
